xla/service/gpu/gpu_latency_hiding_scheduler.cc

```diff
@@ -231,16 +245,36 @@ bool GpuScheduleCrossesOverlapLimit(
       CHECK(
           hlo_query::IsAsyncCollectiveStartOp(curr_hlo_inst.operand(0), true));
       const HloInstruction* curr_start_inst =
-          curr_hlo_inst.operand(0)->async_wrapped_instruction();
+          curr_hlo_inst.opcode() == HloOpcode::kAsyncDone
+              ? curr_hlo_inst.operand(0)->async_wrapped_instruction()
+              : curr_hlo_inst.operand(0);
 
       // If candidate can be overlapped with in-flight collectives
       bool can_overlap = true;
-      for (const auto occupier :
+      for (const auto async_occupier :
            sched_state.resource_occupiers_in_flight.at(resource_type)) {
-        if (sched_state.async_tracker->IsSupportedAsyncStart(*occupier)) {
+        if (sched_state.async_tracker->IsSupportedAsyncStart(*async_occupier)) {
+          HloInstruction* occupier =
+              async_occupier->opcode() == HloOpcode::kAsyncStart
+                  ? async_occupier->async_wrapped_instruction()
+                  : const_cast<HloInstruction*>(async_occupier);
+
           // Number of overlapping ranks between this occupier and candidate
+          std::vector<ReplicaGroup> curr_start_replica_group =
```

Comment:
Can you extract this as utility function to collectives_utils? We might have it already.

Reply:
yea, changed it to use the existing one.